File: TransCoder_tokenized_test_set_functions/python/SORTING_USING_TRIVIAL_HASH_FUNCTION_1.py
def sortUsingHash ( a, n ) :
    Max = max ( a )
    Min = abs ( min ( a ) )
    hashpos = [ 0 ] * ( Max + 1 )
    hashneg = [ 0 ] * ( Min + 1 )
    for i in range ( 0, n ) :
        if a [ i ] >= 0 :
            hashpos [ a [ i ] ] += 1
        else :
            hashneg [ abs ( a [ i ] ) ] += 1
    for i in range ( Min, 0, - 1 ) :
        if hashneg [ i ] != 0 :
            for j in range ( 0, hashneg [ i ] ) :
                print ( ( - 1 ) * i, end = " " )
    for i in range ( 0, Max + 1 ) :
        if hashpos [ i ] != 0 :
            for j in range ( 0, hashpos [ i ] ) :
                print ( i, end = " " )
    

File: TransCoder_tokenized_test_set_functions/cpp/SORTING_USING_TRIVIAL_HASH_FUNCTION_1.cpp
void sortUsingHash ( int a [ ], int n ) {
  int max = * std :: max_element ( a, a + n );
  int min = abs ( * std :: min_element ( a, a + n ) );
  int hashpos [ max + 1 ] = {
    0 };
    int hashneg [ min + 1 ] = {
      0 };
      for ( int i = 0;
      i < n;
      i ++ ) {
        if ( a [ i ] >= 0 ) hashpos [ a [ i ] ] += 1;
        else hashneg [ abs ( a [ i ] ) ] += 1;
      }
      for ( int i = min;
      i > 0;
      i -- ) {
        if ( hashneg [ i ] ) {
          for ( int j = 0;
          j < hashneg [ i ];
          j ++ ) {
            cout << ( - 1 ) * i << " ";
          }
        }
      }
      for ( int i = 0;
      i <= max;
      i ++ ) {
        if ( hashpos [ i ] ) {
          for ( int j = 0;
          j < hashpos [ i ];
          j ++ ) {
            cout << i << " ";
          }
        }
      }
    }